gradle/validation/dependency-analyze.gradle

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This adds gradle-dependency-analyze validation of project dependencies
+
+allprojects { prj ->
+  plugins.withId("java", {
+    prj.apply plugin: 'ca.cutterslade.analyze'
+
+    analyzeClassesDependencies {
+      warnUsedUndeclared = false
+      warnUnusedDeclared = true
```

Comment:
Does this give us a bunch of false positives due to reflective access or something like that?

Reply:
So I'm not a fan of the configurability of this plugin. Basically these are the only configuration options. Either:
* warn = `false` which means fail the build since it will be an error
* warn = `true` which means just print out a warning.

I'm assuming you are referring specifically to `unusedDeclared`. I sadly did not see a way to disable checking for `unusedDeclared`, but only to make it a warning.

Reply:
We could consider something like the approach we have for the spotless plugin, where we select specific modules that it applies to and we turn it off for the rest until they've been migrated.

Reply:
Not sure I understand the suggestion. I don't think there is a way to turn off `warnUnusedDeclared` other than to make it a warning. If you disable the plugin per module, then you lose the `UsedUndeclared` detection as well as far as I can tell.

The PR fixes all modules in the build of `UsedUndeclared`, but leaves the warnings for `UnusedDeclared`